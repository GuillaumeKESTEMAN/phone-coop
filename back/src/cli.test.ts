import { exec } from 'child_process';
import YError from 'yerror';

describe('commands should work', () => {
  it('with ls', async () => {
    const { stdout, stderr } = await execCommand(
      'npm run whook --silent -- ls',
    );

    expect({
      stdout,
      stderr: stderr.replace(/ ([^ ]+)\/whook\//, ' /whook/'),
    }).toMatchSnapshot();
  });

  it('with env', async () => {
    const { stdout, stderr } = await execCommand(
      'npm run whook --silent -- env --name NODE_ENV',
    );

    expect({
      stdout,
      stderr: stderr.replace(/ ([^ ]+)\/whook\//, ' /whook/'),
    }).toMatchSnapshot();
  });

  it('with config', async () => {
    const { stdout, stderr } = await execCommand(
      'npm run whook --silent -- config --name HOST',
    );

    expect({
      stdout,
      stderr: stderr.replace(/ ([^ ]+)\/whook\//, ' /whook/'),
    }).toMatchSnapshot();
  });
});

async function execCommand(
  command: string,
): Promise<{ stdout: string; stderr: string }> {
  return new Promise((resolve, reject) => {
    exec(command, (err, stdout, stderr) => {
      if (err) {
        reject(YError.wrap(err, 'E_COMMAND_FAILURE', stdout, stderr));
        return;
      }
      resolve({ stdout, stderr });
    });
  });
}
