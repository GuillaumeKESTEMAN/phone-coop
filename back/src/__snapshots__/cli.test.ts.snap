// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands should work with config 1`] = `
Object {
  "stderr": "âš¡ - Loading configurations from /whook/packages/whook-example/dist/config/test/config\\".
âœ” - Found the project dir \\"/home/nfroidure/nfroidure/whook/packages/whook-example\\".
",
  "stdout": "\\"localhost\\"
",
}
`;

exports[`commands should work with env 1`] = `
Object {
  "stderr": "âš¡ - Loading configurations from /whook/packages/whook-example/dist/config/test/config\\".
âœ” - Found the project dir \\"/home/nfroidure/nfroidure/whook/packages/whook-example\\".
ðŸ–¥ - Using local env.
",
  "stdout": "test
",
}
`;

exports[`commands should work with ls 1`] = `
Object {
  "stderr": "âš¡ - Loading configurations from /whook/packages/whook-example/dist/config/test/config\\".
âœ” - Found the project dir \\"/home/nfroidure/nfroidure/whook/packages/whook-example\\".
",
  "stdout": "

# Provided by \\"@whook/example\\": 3 commands
- generateOpenAPISchema: Write openAPI schema to stdout
- generateOpenAPITypes: Write openAPI types to stdout
- printEnv: A command printing every env values


# Provided by \\"@whook/cli\\": 6 commands
- config: A simple program that returns the queryed config value
- create: A command helping to create new Whook files easily
- env: A command printing env values
- handler: Runs the given server handler for testing purpose
- inspect: A simple program that returns the result of the injected service
- ls: Print available commands


# Provided by \\"@whook/whook\\": none


# Provided by \\"@whook/cors\\": none
",
}
`;
